ginga/misc/plugins/MultiDim.py

```diff
@@ -399,24 +371,14 @@ def set_hdu(self, idx):
 
         # Nope, we'll have to load it
         self.logger.debug("HDU %d not in memory; refreshing from file" % (idx))
-
-        if info['htype'].lower() in ('bintablehdu', 'tablehdu'):
-            image = AstroTable.AstroTable(logger=self.logger)
-
-        else:
-            # inherit from primary header?
-            inherit_prihdr = self.fv.settings.get('inherit_primary_header',
-                                                  False)
-
-            image = AstroImage.AstroImage(logger=self.logger,
-                                          inherit_primary_header=inherit_prihdr)
-
-        self.image = image
```

Comment:
I like how this collapses nicely and the plugin no longer has to figure out what kind of object to create.